src/binary/ibc-setup/commands/channel.spec.ts

```diff
@@ -0,0 +1,69 @@
+import fs from 'fs';
+
+import test from 'ava';
+import sinon from 'sinon';
+
+import { Link } from '../../../lib/link';
+import { TestLogger } from '../../../lib/testutils';
+import { Logger } from '../../create-logger';
+import { signingClient } from '../../utils/signing-client';
+
+import { simappChain, wasmdChain } from './chains';
+import { Options, run } from './channel';
+
+const fsReadFileSync = sinon.stub(fs, 'readFileSync');
+const consoleLog = sinon.stub(console, 'log');
+
+const mnemonic =
+  'enlist hip relief stomach skate base shallow young switch frequent cry park';
+
+const registryYaml = `
+version: 1
+
+chains:
+  local_wasm:
+    chain_id: testing
+    prefix: wasm
+    gas_price: 0.025ucosm
+    rpc:
+      - http://localhost:26659
+  local_simapp:
+    chain_id: simd-testing
+    prefix: cosmos
+    gas_price: 0.025umuon
+    rpc:
+      - http://localhost:26658`;
+
+test.beforeEach(() => {
+  sinon.reset();
+});
+
+test.serial('creates channel for given connections and ports', async (t) => {
+  const logger = new TestLogger();
+
+  const ibcClientSimapp = await signingClient(simappChain, mnemonic);
+  const ibcClientWasm = await signingClient(wasmdChain, mnemonic);
+  const link = await Link.createWithNewConnections(
+    ibcClientWasm,
+    ibcClientSimapp
+  );
+
+  const options: Options = {
+    home: '/home/user',
+    mnemonic,
+    src: 'local_wasm',
+    dest: 'local_simapp',
+    srcConnection: link.endA.connectionID,
+    destConnection: link.endB.connectionID,
+    srcPort: 'transfer',
+    destPort: 'custom',
+    ordered: false,
+    version: 'ics20-1',
+  };
+
+  fsReadFileSync.returns(registryYaml);
+
+  await run(options, (logger as unknown) as Logger);
+
+  t.assert(consoleLog.calledWithMatch(/Created channels/));
+});
```

Comment:
Not sure what else we could assert here.

Also, is one test case enough?

Reply:
One test is fine.

If the return value were easily parseable, I would say to query that port/channel on both sides.
Maybe a good reason to make the output nicely parseable.

Reply:
386339c